pageserver/src/tenant/timeline.rs

```diff
@@ -3946,13 +3956,26 @@ impl Timeline {
                 LayerResidenceStatus::Resident,
                 LayerResidenceEventReason::LayerCreate,
             );
-            self.insert_historic_layer(x, &mut updates, mapping);
+            if mapping.contains(&x) {
+                duplicated_layers.insert(x.layer_desc().key());
+                warn!(
+                    "duplicated layers generated in compaction: {:?}",
+                    x.layer_desc().key() // TODO: change to use Display
+                );
```

Comment:
```suggestion
                warn!(
                    "duplicated layers generated in compaction: {l:?}"
                );
```

Wouldn't just this work? Unsure why `x` does not shadow `l` since it's the same thing, and `l` is now moved.

Reply:
Unsure what happened to this?

Reply:
fixed by using `{l}` https://github.com/neondatabase/neon/pull/4696/commits/53d0df80f42367fde0352f2caf78acbef41df434